src/reporters/LambdaReporter.ts

```diff
@@ -0,0 +1,104 @@
+import { HttpClient } from "@actions/http-client";
+import { throws } from "assert";
+import { CheckOptions } from "../check";
+import { Result } from "../ComplianceChecker";
+import { Product } from "../model/Product";
+import { Reporter } from "./Reporter";
+
+interface ResultRecord {
+  level: string,
+  rule?: string,
+  check?: string,
+  mandatory?: string,
+  result: string,
+  reason?: string
+}
+
+interface RequestBody {
+  repo_owner?: string,
+  repo_name?: string,
+  start_time?: string,
+  end_time?: string,
+  result?: ResultRecord[]
+}
+
+export class LambdaReporter extends Reporter {
+
+  private readonly uri: string;
+  private readonly keyHeader: any;
+  private readonly http: HttpClient;
+  private body: RequestBody;
+
+  constructor() {
+    super();
+    if (process.env.INPUT_ENG_COMPLIANCE_LAMBDA_URI == null) throw new Error('INPUT_ENG_COMPLIANCE_LAMBDA_URI must be specified');
+    if (process.env.INPUT_ENG_COMPLIANCE_LAMBDA_KEY == null) throw new Error('INPUT_ENG_COMPLIANCE_LAMBDA_KEY must be specified');
+
+    this.uri = process.env.INPUT_ENG_COMPLIANCE_LAMBDA_URI as string;
+    this.body = {};
+    this.http = new HttpClient('eng-compliance-github');
+
+    let key = process.env.INPUT_ENG_COMPLIANCE_LAMBDA_KEY as string;
```

Comment:
<details>
<summary>:warning: <strong>This code uses `let` or `var` to declare one or more variables that are never reassigned.</strong> (click for details)</summary>

A possible improvement would be to declare such a variable with `const` instead.   
  
This change would ensure that it would not be reassigned unintentionally, which could lead to difficulty to comprehend bugs.

**[More details](https://confluence.devfactory.com/x/UA_kEw)**

## The improvement suggestion in this case
```suggestion  
    const key = process.env.INPUT_ENG_COMPLIANCE_LAMBDA_KEY as string;
``` 
> NOTE: Consider applying the structure of this solution in other parts of the code, if there is another occurrence.
</details>

[//]: # (<a id="9c51e063-027d-4ee4-aab1-7a2024d80939" data-finding-id="329518928" href="none" codereview-uuid-reserved-tag data-business-id="55ce652a-40cf-48e1-916a-f8d32a3b8762" data-analyzer="js" />)